### ISSUE TITLE
Hyperparameter optimization for fastText backend

### ISSUE THREAD

**Issue by osma** [NatLibFi/Annif#433]
Implement support for the `hyperopt` command for the fastText backend. Part of #240.

It's probably a bit tricky to come up with a search space that is general enough for different purposes, yet narrow enough that good solutions can be found in reasonable time. There are also pitfalls like the choice of loss function - `softmax` is good but extremely slow on large vocabularies, so it should be possible to force the use of another loss function such as `hs`. Perhaps there should be a way to narrow down the search space already during the hyperopt operation, e.g.

    annif hyperopt -b fasttext-en.loss=hs fasttext-en validation-corpus/
